src/petab_gui/controllers/utils.py

```diff
@@ -17,4 +19,52 @@ def prompt_overwrite_or_append(controller):
     elif msg_box.clickedButton() == overwrite_button:
         return "overwrite"
     elif msg_box.clickedButton() == append_button:
-        return "append"
\ No newline at end of file
+        return "append"
+
+
+class RecentFilesManager(QObject):
+    """Manage a list of recent files."""
+    open_file = Signal(str)  # Signal to open a file
+
+    def __init__(self, max_files=10):
+        super().__init__()
+        self.max_files = max_files
+        # TODO: link together with other settings, i.e. move settings to mc
+        self.settings = QSettings("PEtab_GUI", "PEtab_GUI")
+        self.recent_files = self.load_recent_files()
+        self.tool_bar_menu = QMenu("Recent Files")
+        self.update_tool_bar_menu()
+
+    def add_file(self, file_path):
+        """Add a file to the recent files list."""
+        if file_path in self.recent_files:
+            self.recent_files.remove(file_path)
+        self.recent_files.insert(0, file_path)
+        self.recent_files = self.recent_files[:self.max_files]
+        self.save_recent_files()
+        self.update_tool_bar_menu()
+
+    def load_recent_files(self):
+        """Load recent files from settings."""
+        return self.settings.value("recent_files", [])
+
+    def save_recent_files(self):
+        """Save recent files to settings."""
+        self.settings.setValue("recent_files", self.recent_files)
+
+    def update_tool_bar_menu(self):
+        """Create a menu for the tool bar."""
+        self.tool_bar_menu.clear()
+        for idx, file_path in enumerate(self.recent_files):
+            action = QAction(file_path, self.tool_bar_menu)
```

Comment:
the only comment I have, is whether we really want the full file path in the name, I'd rather have the full path on hover, but something unique and shorter before. But it is fine as is

Reply:
agreed, would be good. will add an issue